SME.CDEP.Aplicacao/Servicos/ServicoCromia.cs

```diff
@@ -0,0 +1,50 @@
+﻿using AutoMapper;
+using SME.CDEP.Aplicacao.DTOS;
+using SME.CDEP.Aplicacao.Servicos.Interface;
+using SME.CDEP.Dominio.Entidades;
+using SME.CDEP.Infra.Dados.Repositorios.Interfaces;
+
+namespace SME.CDEP.Aplicacao.Servicos
+{
+    public class ServicoCromia : IServicoCromia
+    {
+        private readonly IRepositorioCromia repositorioCromia;
+        private readonly IMapper mapper;
+        
+        public ServicoCromia(IRepositorioCromia repositorioCromia, IMapper mapper) 
+        {
+            this.repositorioCromia = repositorioCromia ?? throw new ArgumentNullException(nameof(repositorioCromia));
+            this.mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+        }
+
+        public async Task<long> Inserir(CromiaDTO cromiaDTO)
+        {
+            var cromia = mapper.Map<Cromia>(cromiaDTO);
+            return await repositorioCromia.Inserir(cromia);
+        }
+
+        public async Task<IList<CromiaDTO>> ObterTodos()
+        {
+            return (await repositorioCromia.ObterTodos()).ToList().Where(w=> !w.Excluido).Select(s=> mapper.Map<CromiaDTO>(s)).ToList();
```

Comment:
o primeiro .ToList() não faz sentido pois o where e select geram um IEnumerable novamente.
```suggestion
        public async Task<IList<CromiaDTO>> ObterTodos()
        {
            return (await repositorioCromia.ObterTodos()).Where(w=> !w.Excluido).Select(s=> mapper.Map<CromiaDTO>(s)).ToList();
```